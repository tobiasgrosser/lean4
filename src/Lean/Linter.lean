import Lean.Linter.Util
import Lean.Linter.Builtin
import Lean.Linter.Deprecated
import Lean.Linter.UnusedVariables
import Lean.Linter.MissingDocs
