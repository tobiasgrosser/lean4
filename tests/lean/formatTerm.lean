import Lean
open Lean

def fmt (stx : CoreM Syntax) : CoreM Format := stx >>= PrettyPrinter.formatTerm

#eval fmt `(if c then do t else e)
#eval fmt `(if c then do t; t else e)
#eval fmt `(if c then do t else do e)
#eval fmt `(if let x := c then do t else do e)
#eval fmt `(if c then do t else if c then do t else do e) -- FIXME: make this cascade better?
#eval fmt `(do if c then t else e)
#eval fmt `(do if c then t else if c then t else e)
