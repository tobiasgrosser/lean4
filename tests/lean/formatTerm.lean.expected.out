if c.1 then do
  t.1
else e.1
if c.1 then do
  t.1;
  t.1
else e.1
if c.1 then do
  t.1
else do
  e.1
if let x.1 := c.1 then do
  t.1
else do
  e.1
if c.1 then do
  t.1
else
  if c.1 then do
    t.1
  else do
    e.1
do
  if c.1 then 
    t.1
  else
    e.1
do
  if c.1 then 
    t.1
  else if c.1 then
    t.1
  else
    e.1
